### ISSUE TITLE
High Memory Consumption

### ISSUE THREAD

**Issue by NikhilRanaSG** [microsoft/vscode-dotnettools#1382]
Type: <b>Bug</b>

Enabling C# Dev Kit extension causes "Microsoft.VisualStudio.Code.ServiceHost" to start consuming memory and it keeps growing reaching around 22 GB on my 32 GB laptop. This is causing extreme issues as the device starts to slow down.

Extension version: 1.9.55
VS Code version: Code 1.92.2 (fee1edb8d6d72a0ddff41e5f71a671c23ed924b9, 2024-08-14T17:29:30.058Z)
OS version: Windows_NT x64 10.0.22631
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 9 5900HX with Radeon Graphics         (16 x 3294)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|undefined|
|Memory (System)|31.40GB (11.74GB free)|
|Process Argv| <path> --crash-reporter-id b1de138b-0d26-4cce-abe1-4bd97f6aa69c|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805cf:30301675
binariesv615:30325510
vsaa593:30376534
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
2i9eh265:30646982
962ge761:30959799
pythongtdpath:30769146
welcomedialogc:30910334
pythonnoceb:30805159
asynctok:30898717
pythonregdiag2:30936856
pythonmypyd1:30879173
2e7ec940:31000449
pythontbext0:30879054
accentitlementsc:30995553
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
dsvsc021:30996838
0ee40948:31013168
pythoncenvpt:31062603
a69g1124:31058053
dvdeprecation:31068756
dwnewjupytercf:31046870
2f103344:31071589
impr_priority:31102340
nativerepl1:31104043
refactort:31108082
pythonrstrctxt:31112756
wkspc-onlycs-c:31111717
wkspc-ranged-t:31118572
fje88620:31121564

```

</details>

**Comment by NikhilRanaSG** [comment]
Edited version from v1.10.10 to v1.9.55 as issues was found on v1.9.55. Don't think its happening on v1.10.10 (pre-release).

**Comment by AbhitejJohn** [comment]
@NikhilRanaSG : Thanks for letting us know. It sounds like this isn't reproing on the latest version.

@arkalyanms : Do we have any know issues around memory consumption that might have been fixed in the recent versions of the extension? I do not see anything [here](https://github.com/microsoft/vscode-dotnettools/issues/1274) that might suggest that.

**Comment by arkalyanms** [comment]
@AArnott would you first scout this from the extension perspective before we direct it to a particular team?

**Comment by arkalyanms** [comment]
22G memory leak in servicehost from Lifeng and Andrew's first scout. Sending to project team first.

**Comment by harleend** [comment]
@NikhilRanaSG - Could you share the steps to reproduce this issue or provide more details about the scenario under which you observed the increased memory usage?
From what you've mentioned before, it appears you haven't encountered this problem with the more recent versions. Is that still the case?

**Comment by NikhilRanaSG** [comment]
@harleend 

There are no exact repro steps for it. It just happens when these extensions are installed.

- I came across this issue when using Unity Engine. 
- I have Unity extension installed which also installs 3 other extensions as dependencies.
- I saw my RAM increasing continuously going upto 22 GB for "Microsoft.VisualStudio.Code.ServiceHost".
- Switching to pre-release version fixed this issue for me.

![image](https://github.com/user-attachments/assets/2c056a9e-ae1e-474d-89a2-91c28785afdf)
![image](https://github.com/user-attachments/assets/9c275518-52d4-40b2-a7c5-0f1f860c6c2f)
